apps/prairielearn/src/sync/syncFromDisk.ts

```diff
@@ -43,6 +48,39 @@ export async function syncDiskToSqlWithLock(
   const courseData = await perf.timed('loadCourseData', () =>
     courseDB.loadFullCourse(courseId, courseDir),
   );
+
+  if (config.checkSharingOnSync) {
+    const sharedQuestions = await sqldb.queryRows(
+      sql.select_shared_questions,
+      { course_id: courseId },
+      z.object({
+        id: IdSchema,
+        qid: z.string(),
+      }),
+    );
+    const qids = Object.entries(courseData.questions).map(([qid, _]) => qid);
```

Comment:
Any reason not to do this?

```suggestion
    const qids = Object.keys(courseData.questions);
```

That said, given that `courseData.questions` is already an object and thus has very nice $O(1)$ lookup of keys, why turn this into an array in the first place?

Reply:
ya that's a good point, I can just leave it as an object